addons/account_peppol/models/res_partner.py

```diff
@@ -29,31 +29,45 @@ class ResPartner(models.Model):
     account_peppol_verification_label = fields.Selection(
         selection=[
             ('not_verified', 'Not verified yet'),
-            ('not_valid', 'Not valid'),
+            ('not_valid', 'Not valid'),  # does not exist on Peppol at all
+            ('not_valid_format', 'Cannot receive this format'),  # registered on Peppol but cannot receive the selected document type
             ('valid', 'Valid'),
         ],
         string='Peppol endpoint validity',
         compute='_compute_account_peppol_verification_label',
         copy=False,
-    ) # field to compute the label to show for partner endpoint
+    )  # field to compute the label to show for partner endpoint
 
-    @api.depends('peppol_eas', 'peppol_endpoint', 'ubl_cii_format')
-    def _compute_account_peppol_is_endpoint_valid(self):
-        for partner in self:
-            partner.button_account_peppol_check_partner_endpoint()
+    @api.model
+    def fields_get(self, allfields=None, attributes=None):
+        res = super().fields_get(allfields, attributes)
 
-    @api.depends('account_peppol_is_endpoint_valid', 'account_peppol_validity_last_check')
-    def _compute_account_peppol_verification_label(self):
-        for partner in self:
-            if not partner.account_peppol_validity_last_check:
-                partner.account_peppol_verification_label = 'not_verified'
-            elif partner.account_peppol_is_endpoint_valid:
-                partner.account_peppol_verification_label = 'valid'
-            else:
-                partner.account_peppol_verification_label = 'not_valid'
+        # the orm_cache does not contain the new selections added in stable: clear the cache once
+        peppol_verification_field = self._fields['account_peppol_verification_label']
+        if ('not_valid_format', "Cannot receive this format") not in peppol_verification_field.get_description(self.env)['selection']:
```

Comment:
`get_description` triggers an extra query. You need to check from `res` first to know if you need the update or not